tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Logging/Log4Net/DirectSubmission/LoggingEventDuckBase.cs

```diff
@@ -0,0 +1,53 @@
+﻿// <copyright file="LoggingEventDuckBase.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections;
+using System.ComponentModel;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Logging.Log4Net.DirectSubmission
+{
+    /// <summary>
+    /// Duck type for LoggingEvent
+    /// </summary>
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public abstract class LoggingEventDuckBase
+    {
+        /// <summary>
+        /// Gets the name of the logger that logged the event.
+        /// </summary>
+        public virtual string LoggerName { get; }
+
+        /// <summary>
+        /// Gets the Level of the logging event
+        /// </summary>
+        public virtual LevelDuck Level { get; }
+
+        /// <summary>
+        /// Gets the application supplied message.
+        /// </summary>
+        public virtual object MessageObject { get; }
+
+        /// <summary>
+        /// Gets the message, rendered through the RendererMap".
+        /// </summary>
+        public virtual string RenderedMessage { get; }
+
+        /// <summary>
+        /// Gets the exception object used to initialize this event
+        /// </summary>
+        public virtual Exception ExceptionObject { get; }
+
+        /// <summary>
+        /// Get all the composite properties in this event
+        /// </summary>
+        /// <returns>Dictionary containing all the properties</returns>
+        public virtual IDictionary GetProperties()
+        {
+            return null;
```

Comment:
Would it be safer to return an empty dictionary here just in case? Or is there a reason it's null?

Reply:
This is a duck type, so this method body will never be called, and for that I like returning null as it's more obvious. If a future version removes the method, the duck typing will fail, so it's definitely not being called.

That said, I'll change it if it makes you feel better as it won't make a difference to anything I don't think!😄

Reply:
If we have examples elsewhere I think I am okay with it.

Reply:
Updated to use interfaces instead anyway, so no longer applies